packages/api/src/util/filterEvents.ts

```diff
@@ -14,7 +14,8 @@ export function filterEvents (txHash: H256, { block: { extrinsics, header } }: S
           phase.isApplyExtrinsic &&
           phase.asApplyExtrinsic.eqn(txIndex)
         ),
-        txIndex
+        txIndex,
+        blockNumber: header.number.unwrap()
```

Comment:
As per the linting comments, we just need to move this up.

I'm just a bit concerned that some chains may be configured with `Compact<BlockNumber>` and others with `BlkockNumber`. So just add a bit more protection around this, here is an example used in the derives - https://github.com/polkadot-js/api/blob/f9df7a5c0df68b688b151296967599d8b0b595af/packages/api-derive/src/util/blockNumber.ts#L12-L14

Reply:
Ah I see, thank you. I've pushed a new commit with eslint issues addressed plus adding the guard.

Btw, I wasn't sure if it was a good idea to import the `api-derive/unwrapBlockNumber` function in there so I used the `isCompact` directly instead. Let me know if you want me change it around

Reply:
Happy as-is with the way you structured it.